src/main.cc

```diff
@@ -813,6 +818,18 @@ main(int argc, char **argv)
 		case 'F':
 			box_is_force_recovery = true;
 			break;
+		case 'I':
+#if ENABLE_INTEGRITY
+			opt_mask |= O_INTEGRITY;
+			instance.hashes = (const char *)xstrdup(optarg);
```

Comment:
There is no need in `xstrdup` here: it is used for `name`/`config` to make it consistent with `getenv_safe`, hence `instance.*` fields are released below. So, I suggest to just copy the pointer, but if the contents are required to be copied, please release the copied content.

Reply:
Right, fixed that.